src/Controller/ParticipantDetailsController.php

```diff
@@ -231,4 +262,56 @@ public function participantConsent(
         $response->headers->set('Content-Type', 'application/pdf');
         return $response;
     }
+
+    /**
+     * @Route("/participant/{id}/incentive/{incentiveId}", name="participant_incentive", defaults={"incentiveId": null})
+     */
+    public function participantIncentiveAction(
+        $id,
+        $incentiveId,
+        Request $request,
+        LoggerService $loggerService,
+        EntityManagerInterface $em,
+        ParticipantSummaryService $participantSummaryService,
+        SiteService $siteService
+    ): Response {
+        $participant = $participantSummaryService->getParticipantById($id);
+        $incentive = $incentiveId ? $em->getRepository(Incentive::class)->find($incentiveId) : null;
```

Comment:
Shouldn't this also check that the existing incentive is associated with the participant?